packages/financial-templates-lib/src/price-feed/CoinGeckoPriceFeed.js

```
@@ -0,0 +1,158 @@
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== undefined && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "CoinGeckoPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTimes + this.minTimeBetweenUpdates - currentTime
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "CoinGeckoPriceFeed",
+      message: "Updating CoinGeckoPriceFeed",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime
+    });
+
+    // 1. Construct URL.
+    // See https://www.coingecko.com/api/documentations/v3#/operations/simple/get_simple_token_price__id_ for how this url is constructed.
+    const url =
+      "https://api.coingecko.com/api/v3/simple/token_price/ethereum" +
+      `?contract_addresses=${this.contractAddress}&vs_currencies=${this.currency}`;
+
+    // 2. Send request.
+    const response = await this.networker.getJson(url);
+
+    // 3. Check response.
+    if (!response || !response[this.contractAddress] || !response[this.contractAddress][this.currency]) {
+      throw new Error(`🚨Could not parse result from url ${url}: ${JSON.stringify(response)}`);
+    }
+
+    // 4. Parse result.
+    // Return data structure:
+    // {
+    //   "<contractAddress>": {
+    //     "<currency>": <price>
+    //   }
+    // }
+    const newPrice = this.convertPriceFeedDecimals(response[this.contractAddress][this.currency]);
+
+    // 5. Store results.
+    this.currentPrice = newPrice;
+    this.lastUpdateTime = currentTime;
+    this.priceHistory.push({ time: currentTime, price: newPrice });
+  }
+
+  getCurrentPrice() {
+    return this.invertPrice ? this._invertPriceSafely(this.currentPrice) : this.currentPrice;
+  }
+
+  getHistoricalPrice(time) {
```

This should be async now. Look at the [CryptowatchPriceFeed](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js#L69) for an example implementation of an async getHistoricalPrice.

Additionally, instead of returning `undefined`, `getHistoricalPrice` should throw an error where a price can't be found. See the same example implementation.

---

@nicholaspai we have updated the implementation (including tests) to match the updated interface. Please see https://github.com/UMAprotocol/protocol/pull/2480/commits/05822407a7026f1fbfac48b2627c935c84961c61